### PR TITLE
sectors update-state checks if sector exists before changing its state

### PR DESCRIPTION
**Pull request by geoff-vball**
Closes issue #7754

### CODE REVIEW

**Comment by geoff-vball** [comment]
Haven't been able to test yet. Will get to it Sunday.

File: cmd/lotus-miner/sectors.go [hunk 1]
```diff
@@ -1648,6 +1648,21 @@ var sectorsUpdateCmd = &cli.Command{
...
+		}
+		var sectorExists = false
+		for _, x := range list {
+			if uint64(x) == id {
+				sectorExists = true
+				break
+			}
+		}
+		if !sectorExists {
+			return xerrors.Errorf("sector %s not found, could not change state", cctx.Args().Get(0))
```
Comment (arajasek) [inline cmd/lotus-miner/sectors.go:1663]:
```suggestion
			return xerrors.Errorf("sector %d not found, could not change state", id)
```
```suggestion
			return xerrors.Errorf("sector %s not found, could not change state", cctx.Args().Get(0))
```

File: cmd/lotus-miner/sectors.go [hunk 2]
```diff
@@ -1648,6 +1648,21 @@ var sectorsUpdateCmd = &cli.Command{
 			return xerrors.Errorf("could not parse sector number: %w", err)
 		}
 
+		list, err := nodeApi.SectorsList(ctx)
+		if err != nil {
+			return xerrors.Errorf("could not get list of sectors: %w", err)
+		}
+		var sectorExists = false
+		for _, x := range list {
```
Comment (arajasek) [inline cmd/lotus-miner/sectors.go:1656]: nit: this variable is usually called `v` for `value`

**Review by arajasek — APPROVED** [review]
LGTM, @magik6k express opinions on the undef state thing.

File: cmd/lotus-miner/sectors.go [hunk 3]
```diff
@@ -1648,6 +1648,21 @@ var sectorsUpdateCmd = &cli.Command{
 			return xerrors.Errorf("could not parse sector number: %w", err)
 		}
 
+		list, err := nodeApi.SectorsList(ctx)
```
Comment (magik6k) [inline cmd/lotus-miner/sectors.go:1651]: Why not `SectorsStatus`? Getting a full list will be somewhat slow for bigger miners.

**Review by arajasek — APPROVED** [review]

**Pull request merged**